modules/core/src/main/java/org/apache/ignite/internal/util/nio/ssl/GridNioSslFilter.java

```diff
@@ -216,8 +213,12 @@ public boolean directMode() {
         try {
             GridNioFutureImpl<?> fut = ses.removeMeta(HANDSHAKE_FUT_META_KEY);
 
-            if (fut != null)
+            if (fut != null) {
```

Comment:
I'm not sure if this condition is relevant to `rejectedSesCnt`.

Reply:
Tests show that it is at least more proper way to count them that it is done now.
But I agree, this needs additional investigation and maybe additional testing. Moved to a separate ticket (GG-35573) and removed these edits from PR.